lib/nero/swap/swap_estimate.rb

```diff
@@ -0,0 +1,68 @@
+require 'date'
+
+module Nero
+  module Swap
+    class SwapEstimate < SimpleDelegator
+      INITIAL_PRIOR = 2e-4
+      REJECTION_THRESHOLD = 1e-07
+      DETECTION_THRESHOLD = 0.95
+
+      def adjust(agent, guess)
+        pl = agent.pl
+        pd = agent.pd
+
+        if guess=="LENS"
+          likelihood = pl
+          likelihood /= (pl*probability + (1-pd)*(1-probability))
+        else
+          likelihood = (1-pl)
+          likelihood /= ((1-pl)*probability + pd*(1-probability))
+        end
+
+        guesses << {"timestamp" => DateTime.now.strftime("%Q"), "user_id" => agent.external_id, "answer" => guess, "probability" => likelihood * probability}
+        self
+      end
+
+      def guesses
+        data["guesses"] ||= []
+      end
+
+      def probability
+        (guesses.last && guesses.last["probability"]) || INITIAL_PRIOR
+      end
+
+      def status
+        case
+        when rejected?
+          :rejected
+        when detected?
+          :detected
+        else
+          :active
+        end
+      end
+
+      def active?
+        status == :active
+      end
+
+      def retired?
+        status != :active
+      end
+
+      def seen_by?(user_ids)
```

Comment:
Unused method argument - `user_ids`. If it's necessary, use `_` or `_user_ids` as an argument name to indicate that it won't be used. You can also write as `seen_by?(*)` if you want the method to accept any arguments but don't care about them.